utils.py

```diff
@@ -275,25 +273,93 @@ def set(self, lru_cache):
 ip_cache = LRUCacheSingleton()
 
 
-def is_ip_per_task(app):
-    """
-    Return whether the application is using IP-per-task.
-    :param app:  The application to check.
-    :return:  True if using IP per task, False otherwise.
-    """
-    return app.get('ipAddress') is not None
+def get_app_networking_mode(app):
+    mode = 'host'
 
+    if app.get('ipAddress'):
+        mode = 'container'
 
-def is_user_network(app):
-    """
-    Returns True if container network mode is set to USER
-    :param app:  The application to check.
-    :return:  True if using USER network, False otherwise.
-    """
-    c = app.get('container', {})
-    return c is not None and c.get('type', '') == 'DOCKER' and \
-        c.get('docker', {})\
-        .get('network', '') == 'USER'
+    _mode = app.get('container', {})\
+               .get('docker', {})\
+               .get('network', '')
+    if _mode == 'USER':
+        mode = 'container'
+    elif _mode == 'BRIDGE':
+        mode = 'container/bridge'
+
+    networks = app.get('networks', [])
+    for n in networks:
+        # Modes cannot be mixed, so assigning the last mode is fine
+        mode = n.get('mode', 'host')
```

Comment:
I don't expect the default `host` to ever kick in here because Marathon would have normalized any items in `networks` to have a `mode`. That said, `host` isn't correct: if an empty network is specified (e.g. `networks: [ {} ]`) then the `mode` of that network is actually `container`

https://github.com/mesosphere/marathon/blob/596e86111e681e9923d7f751894ba62aaee64af5/docs/docs/rest-api/public/api/v2/types/network.raml#L73